src/AutoMapper/Execution/DelegateBasedResolver.cs

```diff
@@ -4,30 +4,29 @@ namespace AutoMapper.Execution
 {
     using System;
 
-    public class DelegateBasedResolver<TSource, TMember> : IMemberResolver
+    public class DelegateBasedResolver<TSource, TMember> : IValueResolver
```

Comment:
This exposed the type so PropertyMap.SourceType would use it. I presume now works differently.